src/runtime_src/core/pcie/tools/xbutil/xbutil.cpp

```diff
@@ -906,33 +907,55 @@ int runShellCmd(const std::string& cmd, std::string& output)
     return ret;
 }
 
+int searchXsaAndDsa(std::string xsaPath, std::string 
+    dsaPath,struct stat &st, std::string& path, std::string &output) 
+{
+    if (stat(xsaPath.c_str(), &st) != 0) {
+            if (stat(dsaPath.c_str(), &st) != 0) {
+                output += "ERROR: Failed to find test in ";
+                output += xsaPath;
+                output += " and ";
+                output += dsaPath;
+                return -ENOENT;
+            }
+            path =  dsaPath;
+            return EXIT_SUCCESS;
+        } else {
+            path = xsaPath;
+            return EXIT_SUCCESS;
+        }
+}
+
 int xcldev::device::runTestCase(const std::string& exe,
-    const std::string& xclbin, std::string& output)
+    const std::string& xclbin, const bool isPython, std::string& output)
```

Comment:
there is no need to pass in isPython since you do the search internally anyway. You should know what exe to use based on the search result.